packages/js-dash-sdk/src/SDK/Client/Platform/Fetcher/Fetcher.ts

```diff
@@ -0,0 +1,168 @@
+import DAPIClient from '@dashevo/dapi-client';
+import { Identifier } from '@dashevo/wasm-dpp/dist';
+import {
+  GetDataContractResponse,
+  GetIdentityResponse,
+} from '@dashevo/dapi-grpc/clients/platform/v0/web/platform_pb';
+
+import NotFoundError from '@dashevo/dapi-client/lib/transport/GrpcTransport/errors/NotFoundError';
+import { GetDocumentsResponse } from '@dashevo/dapi-client/lib/methods/platform/getDocuments/GetDocumentsResponse';
+import withRetry from './withRetry';
+import { FetchOpts } from '../methods/documents/get';
+
+type FetcherOptions = {
+  /**
+   * Multiplier for delay between retry attempts
+   */
+  delayMulMs?: number;
+  /**
+   * Maximum number of retry attempts
+   */
+  maxAttempts?: number;
+};
+
+const DEFAULT_DELAY_MUL_MS = 1000;
+const DEFAULT_MAX_ATTEMPTS = 7;
+
+/**
+ * Fetcher class that handles retry attempts for acknowledged identifiers
+ * Primary goal of this class is to mitigate network propagation lag
+ * where we query platform entities right after their creation
+ *
+ * Should be used until fully functioning state transition acknowledgement is implemented
+ */
+class Fetcher {
```

Comment:
Won't it be cleaner if you make it abstract and then extend it for each type with `fetch` method implementation?

Reply:
Agreed to do not spend time on this temporary solution